src/Topshelf/Runtime/Windows/WindowsServiceHost.cs

```diff
@@ -237,6 +238,29 @@ protected override void OnSessionChange(SessionChangeDescription changeDescripti
             }
         }
 
+        protected override bool OnPowerEvent(PowerBroadcastStatus powerStatus)
+        {
+            try
+            {
+                _log.Info("[Topshelf] Power event changed");
+
+                var arguments = new WindowsPowerEventArguments(powerStatus);
+
+                var result = this._serviceHandle.PowerEvent(this, arguments);
+
+                _log.Info("[Topshelf] Stopped");

```

Comment:
The service stops? Is that the only outcome from these events -- because I don't see anything indicating a shutdown cycle.